internal/bandwidth_routing.go

```diff
@@ -50,45 +47,35 @@ func chooseNextHop(weights []float64,
 
 func newBandwidthRouting(r *reachabilityHandler) *bandwidthRouting {
 	return &bandwidthRouting{
-	  r,
-		NewBandwidthEstimator(),
-		sync.RWMutex{},
+		r,
+		nil,
 	}
 }
 
 // Finds the next place to send a packet.
 // See the routingAlgorithm interface for details.
 func (b *bandwidthRouting) FindNextHop(id types.NodeAddress,
-      src types.NodeAddress) (types.NodeAddress, error) {
-  possible_next, err := b.reachability.FindPossibleDests(id, src)
+	src types.NodeAddress) (types.NodeAddress, error) {
+	possible_next, err := b.reachability.FindPossibleDests(id, src)
 	if err != nil {
 		return "", err
 	}
 
 	// Decide which one of our possible destinations to send it to.
-	b.bandwidth_mutex.RLock()
 	weights := b.bandwidths.GetWeights(possible_next)
-	b.bandwidth_mutex.RUnlock()
 
 	// Choose a random destination given our weights.
 	return chooseNextHop(weights, possible_next), nil
 }
 
-// Send a packet. See routingAlgorithm interface for details.
-func (b *bandwidthRouting) SendPacket(dest DataConnection,
-                                      addr types.NodeAddress,
-                                      packet types.Packet) error {
-	b.bandwidth_mutex.Lock()
-	b.bandwidths.WillSendPacket(addr)
-	b.bandwidth_mutex.Unlock()
+// Sets the routing handler that we will use with this algorithm.
+// See the routingAlgorithm interface for details.
+func (b *bandwidthRouting) BindToRouting(routing *routingHandler) {
```

Comment:
Can you just pass in the bandwidth estimator at construction time / always create it?